docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -567,30 +589,30 @@ Let's suppose we have a template called `item.html`:
 <1> `showImage` is a named parameter.
 <2> `it` is a special key that is replaced with the first unnamed param of the tag.
 
-Now if we register this template under the name `item` and if we add a `UserTagSectionHelper` to the engine:
+Now if we register this template under the name `itemDetail` and if we add a `UserTagSectionHelper` to the engine:
 
 [source,java]
 ----
 Engine engine = Engine.builder()
-                   .addSectionHelper(new UserTagSectionHelper.Factory("item"))
+                   .addSectionHelper(new UserTagSectionHelper.Factory("myItem","itemDetail.html"))
                    .build();
 ----
 
-NOTE: In Quarkus, all files from the `src/main/resources/templates/tags` are registered and monitored automatically.
+NOTE: In Quarkus, all files from the `src/main/resources/templates/tags` are registered and monitored automatically!
 
 We can include the tag like this:
 
 [source,html]
 ----
 <ul>
-{#each items}
+{#for item in items}
   <li>
-  {#item this showImage=true /} <1>
+  {#myItem item showImage=true /} <1>
```

Comment:
This is the first place I realised how we can pass more than one parameter to a tag. Is this formally documented somewhere else in this file? If not, it should, I was sure I would need a comma.

Reply:
Comma was never required. It's not documented. The whole SPI needs docs but since it's still a WIP it has a lower priority. In short, a section/tag may consume any number of named/unnamed parameters and it's the responsibility of a section to interprete the params, i.e. it could be an expression ('item.name') or it could be ignored (such as `in` in the `{#for item in items}`).

We already have https://github.com/quarkusio/quarkus/issues/7326.

Reply:
So, this is what bugs me about the regularity of those tags. Some use `in` or `as` keywords (not evaluated) while others use `=` which seems to be an operator, but I'm not clear on that either.

The semantics are not clear to me as a user. And it doesn't appear regular.

Compare this with the following syntax:

```
<ul>
#{list items:products, as:'product'}
    <li>${product}</li>
#{/list}
</ul>
```

Where you can pass exactly one unnamed parameter to a tag, or you pass them by name with the `:` operator and `,` separator. As you can see, here `products` is an expression, while `'product'` is a string. The semantics are clear.

It's less flexible, but it's much more consistent with programming language constructs. For example, I'm not sure how the current semantics of passing parameters as-is to Java tag implementation combines with the rule for named parameters: can I call `#{foo first second third=something}` and expect it to work if `foo` is a Java tag?

Using the current Qute syntax, this would mean:

```
{#for items, as='item'}
```

Now, given that we only have `as` in the problematic `with` form (which we should remove in favour of `let`/`set`), that only leaves us with a single keyword `in`, and I've observed people mostly write custom tags using taglet files (not in Java), perhaps there's not much risk in this sort of keybord behaviour proliferating to confuse users.

But at the very least, we need to document that parameters are passed separated by spaces, named parameters use `=`.